### PR TITLE
Safety pr

### PR DESCRIPTION
**Pull request by kyr-pol**
Adding Safety Features and Tool. Current Version works and has been tested, needs cleaning up.

### CODE REVIEW

**Comment by nrontsis** [comment]
I think that this PR consists of the following components:
- `examples/*.py`, `safe_pilco/experiments.py`, `safe_pilco/linear_cars_env.py`, `safe_pilco/safe_cars_run.py`, `safe_pilco/safe_swimmer_run.py`.
These are examples that aim to produce reproducible results with (safe)PILCO on specific tasks. These can move to a different repo.
- Modifications to code already in the repo (as done e.g. for `pilco/controllers.py`, `pilco/mgpr.py` etc.). These appear to be general and it might be better to be split into a separate PR that will be merged before this one.
- All other new files. These aim to bring safety features to this repo.

I suggest that we first deal with the first two bullet points so that we get a better understanding of the amount of code entailed in the last point. Then, we could decide about how this will be handled (single file, separate folder/module etc.)

File: pilco/models/safe_pilco.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from pilco.models import PILCO
```
Comment (nrontsis) [inline pilco/models/safe_pilco.py]: Is this file a duplicate of `safe_pilco/safe_pilco.py`?

**Comment by kyr-pol** [comment]
Ok so added a commit to clean some things up, taking script files etc. out, to a new repo.

Changes as you said consist of:
- Some changes in the existing examples we provide with the repo.
- Mostly slight changes to existing core functions, such as `controllers.py`
- Extra files in the safe_pilco folder that add functionality: `safe_pilco.py` that inherits from pilco.models.pilco and `safe_rewards.py` that provides additional types of reward functions
- Extra files in the safe_pilco folder that add examples: `linear_cars_env.py`, `safe_cars_run.py`, `safe_swimmer_run.py`, that showcase how the new safety features can be used, just as the examples provided in the examples folder do for the rest of the codebase.

I agree with merging in steps so that we can better keep track of what is going on.

**Comment by nrontsis** [comment]
Closing in favour of #45

**Pull request closed without merging**